Twister LLVM builds fail on frdm_k64f due to tf-psa-crypto `\retval` documentation warnings treated as errors

### Describe the bug

Our CI started seeing widespread Twister build failures on `frdm_k64f/mk64f12`
when building Zephyr with LLVM/clang.

The common failure signature is in
`modules/crypto/tf-psa-crypto/include/psa/crypto.h`, where clang reports
`-Wdocumentation` errors such as:

```text
.../modules/crypto/tf-psa-crypto/include/psa/crypto.h:3258:34: error:
empty paragraph passed to '\retval' command [-Werror,-Wdocumentation]
.../modules/crypto/tf-psa-crypto/include/psa/crypto.h:3259:34: error:
empty paragraph passed to '\retval' command [-Werror,-Wdocumentation]
.../modules/crypto/tf-psa-crypto/include/psa/crypto.h:3260:41: error:
empty paragraph passed to '\retval' command [-Werror,-Wdocumentation]
```

This appears to fan out into many unrelated test build failures because the
same header is used broadly.

From the CI email/logs, examples of impacted tests include:

- `bootloader.mcuboot`
- `bootloader.mcuboot.swap_using_move`
- `mgmt.mcumgr.fs.mgmt.hash.supported.sha256`
- `libraries.encoding.jwt.ecdsa.psa`
- `secure_storage.psa.its.secure_storage.store.settings`

All of them fail with the same `tf-psa-crypto/include/psa/crypto.h`
`\retval` documentation error and then stop with `ninja: build stopped:
subcommand failed`.

### Regression

- [x] This is a regression.

The failing Zephyr revision from CI is:
`e304ae53af7f7ea703b5c378711652b645192143`

At that Zephyr revision, the west manifest points `tf-psa-crypto` to:
`efdf9e209cc3b2c7be93ac81c7c96cd2550bbe27`

### Steps to reproduce

A full reproducer from the CI environment is not available in this report, but
this is consistently reproduced by Twister/LLVM builds on:

1. Sync Zephyr at commit
   `e304ae53af7f7ea703b5c378711652b645192143`
2. Update modules from the matching west manifest
3. Run Twister with an LLVM/clang toolchain on `frdm_k64f/mk64f12`
4. Build any affected test that includes `psa/crypto.h`
5. Observe clang failing on `\retval` documentation warnings promoted to
   errors

One representative failure from CI:

```text
bootloader.mcuboot  v4.4.0-3526-ge304ae53af7f  Build failure  frdm_k64f/mk64f12
3259 | * \retval #PSA_ERROR_DATA_INVALID |
.../modules/crypto/tf-psa-crypto/include/psa/crypto.h:3259:34: error:
empty paragraph passed to '\retval' command [-Werror,-Wdocumentation]
3260 | * \retval #PSA_ERROR_CORRUPTION_DETECTED |
.../modules/crypto/tf-psa-crypto/include/psa/crypto.h:3260:41: error:
empty paragraph passed to '\retval' command [-Werror,-Wdocumentation]
~7 errors generated.
ninja: build stopped: subcommand failed.
```

### Relevant log output

```shell
bootloader.mcuboot  v4.4.0-3526-ge304ae53af7f  Build failure  frdm_k64f/mk64f12
3259 | * \retval #PSA_ERROR_DATA_INVALID |
.../modules/crypto/tf-psa-crypto/include/psa/crypto.h:3259:34: error:
empty paragraph passed to '\retval' command [-Werror,-Wdocumentation]
3259 | * \retval #PSA_ERROR_DATA_INVALID |
     |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
3260 | * \retval #PSA_ERROR_CORRUPTION_DETECTED |
.../modules/crypto/tf-psa-crypto/include/psa/crypto.h:3260:41: error:
empty paragraph passed to '\retval' command [-Werror,-Wdocumentation]
3260 | * \retval #PSA_ERROR_CORRUPTION_DETECTED |
     |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
3261 | * \retval #PSA_ERROR_BAD_STATE |
~7 errors generated.
ninja: build stopped: subcommand failed.
FAILED: _sysbuild/sysbuild/images/bootloader/mcuboot-prefix/src/mcuboot-stamp/mcuboot-build
.../twister-out/frdm_k64f_mk64f12/zephyr_llvm/tests/boot/test_mcuboot/
bootloader.mcuboot/mcuboot && /usr/bin/cmake --build .
ninja: build stopped: subcommand failed.
```

### Impact

Major – Severely degrades functionality; workaround is difficult or
unavailable.

This causes a large number of LLVM-based CI builds to fail even though the
underlying problem appears to be limited to documentation markup in a shared
header.

### Environment

- OS: Linux in CI (Jenkins-based environment)
- Toolchain: LLVM/clang (`zephyr_llvm` Twister runs)
- Board/platform: `frdm_k64f/mk64f12`
- Zephyr commit: `e304ae53af7f7ea703b5c378711652b645192143`
- tf-psa-crypto revision from Zephyr west manifest:
  `efdf9e209cc3b2c7be93ac81c7c96cd2550bbe27`

### Additional Context

This came from a CI email for Jenkins job:
`bitbucket_build_platform #9713`

I searched for an existing Zephyr issue with the same failure signature but did
not find an obvious duplicate.

The issue may belong to the `tf-psa-crypto` content itself, but it currently
surfaces as a Zephyr regression because the pinned module revision in the Zephyr
manifest causes broad CI breakage.

--- a/west.yml
+++ b/west.yml
@@ -389,7 +389,7 @@ manifest:
         - testing
         - tee
     - name: tf-psa-crypto
-      revision: efdf9e209cc3b2c7be93ac81c7c96cd2550bbe27
+      revision: 2d881593696aec687ca2246c53de207c2b9bf782
       path: modules/crypto/tf-psa-crypto
       groups:
         - crypto

